lib/ansible/modules/packaging/os/redhat_subscription.py

```diff
@@ -407,7 +407,7 @@ def unregister(self):
             Raises:
               * Exception - if error occurs while running command
         '''
-        args = [SUBMAN_CMD, 'unregister']
+        args = [SUBMAN_CMD, 'remove', '--all']
```

Comment:
Out of curiosity, when was `remove` added? Is there a chance that this will crash on a systems that `remove` isn't a thing yet? In that case it might be better to have a fallback to `unregister`.

Reply:
I agree a fallback would be good. I'll add it of course.

Reply:
so in fact the unregister command was removed in 2013: https://github.com/candlepin/subscription-manager/commit/1a9884877bf833225ea89bd7ff01ee51a5687ee7
Then reverted in 2013 as well: https://github.com/candlepin/subscription-manager/commit/9e0893737882da2fb25163696a0e14cf4b4c112b

Which means both unregister and remove have been supported since 2013... I think we're good to switch to remove now. Let me know what you think.

Reply:
I think switching to `remove --all` is fine. That matches what is in the Red Hat documentation.

We should try `remove --all` first, then fall back to `unregister` if it fails.